integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/dataload/TestGlobalSortDataLoad.scala

```diff
@@ -527,6 +529,73 @@ class TestGlobalSortDataLoad extends QueryTest with BeforeAndAfterEach with Befo
     assert(sql("select * from carbon_global_sort_update").count() == 22)
   }
 
+  test("calculate the global sort partitions automatically when user does not give in load options ") {
```

Comment:
add more test case to config a small defaultMaxSplitBytes, let it has multiple partitions.